pkg/distributor/distributor.go

```diff
@@ -482,6 +476,20 @@ func New(cfg Config, clientConfig ingester_client.Config, limits *validation.Ove
 		subservices = append(subservices, d.ingestStorageWriter)
 	}
 
+	// Register each metric only if the corresponding storage is enabled.
+	// Some queries in the mixin use the presence of these metrics as indication whether Mimir is running with ingest storage or not.
+	if cfg.IngestStorageConfig.Enabled {
+		promauto.With(reg).NewGauge(prometheus.GaugeOpts{
+			Name: "cortex_distributor_ingest_storage_enabled",
+			Help: "Whether writes are being processed via ingest storage. Always 1 if ingest storage is enabled.",
+		}).Set(1)
+	} else {
+		promauto.With(reg).NewGauge(prometheus.GaugeOpts{
+			Name: "cortex_distributor_replication_factor",
+			Help: "The configured replication factor.",
+		}).Set(float64(ingestersRing.ReplicationFactor()))
```

Comment:
I think this deserves a `[CHANGE]` in the changelog.

Reply:
we haven't entered any of the sigyn changes in the changelog, so i wasn't sure what to do here. I can add one

Reply:
Oh, that's a good point. This only applies to ingest-storage. We should gather all changelog entries, and not forget about this once we write them.

Reply:
added in https://github.com/grafana/mimir/pull/7809/commits/1b734cf71e8fbc03d31e73e6f1a8b477d49e3d8f

Reply:
in this case i'll remove the entry. GH didn't show your comment when i left mine